cypress-io/cypress
# feat: add banner to enable system notifications for runs

**Note that this is merging into a feature branch - feature/run-notifications-m1**



- Closes https://github.com/cypress-io/cypress/issues/26781

### Additional details


This PR adds a banner in the app header that allows users to enable system notifications for the Cypress app.

### Steps to test

1. Open a project that has all cloud steps completed (user is logged in, project is connected, at least one run is recorded)
2. Verify that when these conditions are met, you see the banner on all of the pages listed in the ClickUp and that it matches Figma
3. Click to enable notifications and verify that you get a notification telling you that notifications are enabled.
4. Check out the test cases that I added to `packages/app/cypress/e2e/top-nav.cy.ts` that covers all of the other cases for the banner

Links to the ClickUp requirements and Figma mockups are in the related issue

### How has the user experience changed?


<img width="1331" alt="Screenshot 2023-05-24 at 1 58 52 PM" src="https://github.com/cypress-io/cypress/assets/14275198/14224a2b-40ab-4661-a4b4-e7e7e98c26fc">

### Walkthrough
https://www.loom.com/share/b9e2bf8033db48799c64faa4ecee43f7

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
